Added guidelines and template for control definitions. (#271)
finos/common-cloud-controls: docs/governance/community-guidelines/content-standards-and-practices/identifiers.md

# CCC Taxonomy ID Assignment Guidelines

This document provides a standardized approach for assigning unique identifiers (IDs) to various entities within the CCC Taxonomy, including service types, service categories, features, threats, controls, test requirements, and tests. Adhering to this standard ensures consistency and clarity across all related documentation and implementations.

| **ID Type**          | **Format**                                 | **Example**           |
|----------------------|--------------------------------------------|-----------------------|
| **Service Type**     | `CCC.<ServiceType>`                        | `CCC.Storage`         |
| **Service Category** | `CCC.<ServiceCategory>`                    | `CCC.ObjStor`         |
| **Feature**          | `CCC.<ServiceCategory>.F<##>`              | `CCC.ObjStor.F01`     |
| **Threat**           | `CCC.<ServiceCategory>.T<##>`              | `CCC.RDMS.T01`        |
| **Control**          | `CCC.<ServiceCategory>.C<##>`              | `CCC.VM.C01`          |
| **Test Requirement** | `CCC.<ServiceCategory>.C<##>.TR<##>`       | `CCC.VM.C01.TR01`     |

> | **Test**             | `CCC.<ServiceCategory>.C<##>.TR<##>.T<##>` | `CCC.VM.C01.TR01.T01` |

Reviewer: | **Test**             | `CCC.<ServiceCategory>.C<##>.TR<##>.TE<##>` | `CCC.VM.C01.TR01.TE01` |
```

> | **Threat**           | `CCC.<ServiceCategory>.T<##>`              | `CCC.RDMS.T01`        |
> | **Control**          | `CCC.<ServiceCategory>.C<##>`              | `CCC.VM.C01`          |
> | **Test Requirement** | `CCC.<ServiceCategory>.C<##>.TR<##>`       | `CCC.VM.C01.TR01`     |
> | **Test**             | `CCC.<ServiceCategory>.C<##>.TR<##>.T<##>` | `CCC.VM.C01.TR01.T01` |

Author: ```suggestion
| **Threat**           | `CCC.<ServiceCategory>.TH<##>`              | `CCC.RDMS.TH01`        |
| **Control**          | `CCC.<ServiceCategory>.C<##>`              | `CCC.VM.C01`          |
| **Test Requirement** | `CCC.<ServiceCategory>.C<##>.TR<##>`       | `CCC.VM.C01.TR01`     |
| **Test**             | `CCC.<ServiceCategory>.C<##>.TR<##>.TE<##>` | `CCC.VM.C01.TR01.TE01` |
```

## Service Types

Service types are used to group related service categories under a common label.

Each service type ID follows the format `CCC.<ServiceType>`.

### Examples:

- **CCC.Storage** - Represents the Storage service type.
- **CCC.Compute** - Represents the Compute service type.
- **CCC.DB** - Represents the Database service type.
- **CCC.Network** - Represents the Networking service type.

## Service Categories

> Service categories are specific classifications within a service type. Because the service category name is unique, the ID does not need to reference the parent type.

Reviewer 2: Will this cause confusion between what are Service Types / Service Categories? Would we lose some information on the hierarchy? Would this make it harder to navigate/search?

(Just looking for a discussion on these to make sure they have been considered, rather than saying we can't take this approach for these reasons, as I appreciate that it including the service type could just add a lot of noise).

Author: This hasn't been a significant discussion to-date. Happy to hear additional feedback. My primary motivation with this was to reduce the ID length by a few characters, as it's significant once we get down to the lower level entries.

Reviewer: Hi @smendis-scottlogic  - in a working session @eddie-knight and I discussed the pros/cons of concatenating service types and categories. Our conclusion was to not change and leave them as-is as proposed.

Each service category ID follows the format `CCC.<ServiceCategory>`.

### Examples:

- **CCC.ObjStor** - Represents the Object Storage service category under `CCC.Storage`.
- **CCC.RDMS** - Represents the Relational Database Management System service category under `CCC.DB`.
- **CCC.VM** - Represents the Virtual Machine service category under `CCC.Compute`.

## Features

Features are specific functionalities or capabilities that are expected for a service to be portable with other services in that service category.

Each feature ID follows the format `CCC.<ServiceCategory>.F<##>`.

### Examples:

- **CCC.ObjStor.F01** - The first feature in the Object Storage service category.
- **CCC.RDMS.F01** - The first feature in the RDMS service category.

## Threats

Threats are potential security risks associated with a service category.
Controls and threats have a "many to many" releationship, where 

> Each threat ID follows the format `CCC.<ServiceCategory>.T<##>`.

Author: ```suggestion
Each threat ID follows the format `CCC.<ServiceCategory>.TH<##>`.
```

### Examples:

> - **CCC.RDMS.T01** - The first threat in the RDMS service category.
> - **CCC.VM.T10** - The tenth threat in the Virtual Machine service category.

Author: ```suggestion
- **CCC.RDMS.TH01** - The first threat in the RDMS service category.
- **CCC.VM.TH10** - The tenth threat in the Virtual Machine service category.
```

## 5. Controls

Controls are security measures designed to mitigate specific threats.
Each control will contain a set of Test Requirements.

Each control ID follows the format `CCC.<ServiceCategory>.C<##>`.

### Examples:

- **CCC.VM.C01** - The first control in the Virtual Machine service category.
- **CCC.ObjStor.C02** - The second control in the Object Storage service category.

## 6. Test Requirements

Test requirements are specific conditions or criteria that must be met to validate the associated control. 
Each Test Requirement will map to a set of Tests.

Each test requirement ID follows the format `CCC.<ServiceCategory>.C<##>.TR<##>`.

### Examples:

- **CCC.VM.C01.TR01** - The first test requirement for the first control in the Virtual Machine service category.
- **CCC.ObjStor.C02.TR02** - The second test requirement for the second control in the Object Storage service category.

## 7. Tests

Tests are individual assessments or procedures that fulfill a test requirement. These should be written in Gherkin for easy implementation.

> Each test ID follows the format `CCC.<ServiceCategory>.C<##>.TR<##>.T<##>`.

Reviewer: ```suggestion
Each test ID follows the format `CCC.<ServiceCategory>.C<##>.TR<##>.TE<##>`.
```

### Examples:

> - **CCC.VM.C01.TR01.T01** - The first test for the first test requirement of the first control in the Virtual Machine service category.
> - **CCC.ObjStor.C02.TR02.T02** - The second test for the second test requirement of the second control in the Object Storage service category.

Author: ```suggestion
- **CCC.VM.C01.TR01.TE01** - The first test for the first test requirement of the first control in the Virtual Machine service category.
- **CCC.ObjStor.C02.TR02.TE02** - The second test for the second test requirement of the second control in the Object Storage service category.
```
